### PR TITLE
👌🏻  Added optional MyParcel.com client credentials as input for shops setup

### PR DESCRIPTION
**Pull request by yoan-myparcel**
# What's Changed
The _ShopSetupRequest_ can now give MyParcel.com client credentials (when available), and perform basic validation on them.

### CODE REVIEW

File: src/Shop/Http/Requests/ShopSetupRequest.php [hunk 1]
```diff
@@ -11,13 +11,26 @@ class ShopSetupRequest extends FormRequest
     public function rules(): array
     {
         return [
-            'data.settings'     => 'array',
-            'data.redirect_url' => 'url',
+            'data.settings'         => 'array',
+            'data.redirect_url'     => 'url',
```
Comment (daniel-myparcel) [inline src/Shop/Http/Requests/ShopSetupRequest.php:15]: Don't we always need `data.redirect_url`? Consider making the `data.redirect_url` required.

File: tests/Shop/Http/Requests/ShopSetupRequestTest.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    #[DataProvider('success_data')]
+    public function test_validation_rules_pass(): void
+    {
+        $request = new ShopSetupRequest();
+        $rules = $request->rules();
+
+        $validator = new Validator(
+            Mockery::mock(Translator::class),
+            ['data' => []],
```
Comment (daniel-myparcel) [inline tests/Shop/Http/Requests/ShopSetupRequestTest.php:126]: I don't have experience using `#[DataProvider()]` but don't you need to pass `$data` in the validator? (Same as the test below)

**Review by daniel-myparcel — CHANGES_REQUESTED** [review]

File: src/Shop/Http/Requests/ShopSetupRequest.php (hunk 1 above)
Comment (yoan-myparcel) [reply src/Shop/Http/Requests/ShopSetupRequest.php:15]: That is correct!

File: tests/Shop/Http/Requests/ShopSetupRequestTest.php (hunk 2 above)
Comment (yoan-myparcel) [reply tests/Shop/Http/Requests/ShopSetupRequestTest.php:126]: hahaha yeah, no wonder why this tests always worked 🤔

Comment (yoan-myparcel) [reply tests/Shop/Http/Requests/ShopSetupRequestTest.php]: Fixed it, and also found an edge case - `'mp_client' => []` would pass the validation, while it shouldn't.

Comment (daniel-myparcel) [reply tests/Shop/Http/Requests/ShopSetupRequestTest.php]: Noice

**Review by daniel-myparcel — APPROVED** [review]
